content/docs/self-hosted-runners.md

```diff
@@ -1,82 +1,81 @@
+resources, or to train in the cloud.
 
 ☝️ **Tip!** Check out the official documentation from
 [GitHub](https://help.github.com/en/actions/hosting-your-own-runners/about-self-hosted-runners)
-and [GitLab](https://docs.gitlab.com/runner/) to get started setting up your
+and [GitLab](https://docs.gitlab.com/runner) to get started setting up your
 self-hosted runner.
 
-## Allocating cloud resources with CML
+## Allocating Cloud Compute Resources with CML
 
-When a workflow requires computational resources (such as GPUs) CML can
+When a workflow requires computational resources (such as GPUs), CML can
 automatically allocate cloud instances using `cml-runner`. You can spin up
-instances on your AWS or Azure account (GCP support is forthcoming!).
+instances on your [AWS](#aws), [Azure](#azure), [GCP](#gcp), or
+[Kubernetes](#kubernetes) accounts.
 
 For example, the following workflow deploys a `t2.micro` instance on AWS EC2 and
 trains a model on the instance. After the job runs, the instance automatically
-shuts down. You might notice that this workflow is quite similar to the
-[basic use case](/doc/usage) highlighted in the beginning of the docs- that's
-because it is! What's new is that we've added `cml-runner`, plus a few
-environmental variables for passing your cloud service credentials to the
+shuts down.
+
+You might notice that this workflow is quite similar to the
+[basic use case](/doc/usage). The only addition is `cml-runner` and a few
+environment variables for passing your cloud service credentials to the
 workflow.
 
+Note that `cml-runner` will also automatically restart your jobs (whether from a
+[GitHub Actions 72-hour timeout](https://docs.github.com/en/actions/reference/usage-limits-billing-and-administration#usage-limits)
+or a
+[AWS EC2 spot instance interruption](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/spot-interruptions.html)).
+
 ```yaml
-name: 'Train-in-the-cloud'
+name: Train-in-the-cloud
 on: [push]
-
 jobs:
   deploy-runner:
-    runs-on: [ubuntu-latest]
+    runs-on: ubuntu-latest
     steps:
       - uses: iterative/setup-cml@v1
       - uses: actions/checkout@v2
-      - name: 'Deploy runner on EC2'
-        shell: bash
+      - name: Deploy runner on EC2
         env:
           REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
         run: |
           cml-runner \
-          --cloud aws \
-          --cloud-region us-west \
-          --cloud-type=t2.micro \
-          --labels=cml-runner
-  model-training:
+              --cloud aws \
+              --cloud-region us-west \
+              --cloud-type t2.micro \
+              --labels cml-runner
+  train-model:
     needs: deploy-runner
     runs-on: [self-hosted, cml-runner]
-    container: docker://iterativeai/cml:0-dvc2-base1
+    container: docker://iterativeai/cml:0-dvc2-base1-gpu
```

Comment:
```suggestion
    container: docker://iterativeai/cml:0-dvc2-base1
```

This is **not** a GPU example: see `--cloud-type t2.micro` above; that instance doesn't provide any accelerator. If you hint anywhere that it does support GPU, users can be heavily confused because it's lacking the required Docker options to make it work on GPU instances:

```suggestion
    container:
      image: docker://iterativeai/cml:0-dvc2-base1-gpu
      options: --gpus all
```

Reply:
better to give a GPU example here. Changed all text refs too.